src/AddressRegistry/StreetName/StreetName.cs

```diff
@@ -328,15 +334,21 @@ public void CorrectAddressBoxNumber(AddressPersistentLocalId addressPersistentLo
         public void RemoveAddress(AddressPersistentLocalId addressPersistentLocalId)
         {
             StreetNameAddresses
-                .GetByPersistentLocalId(addressPersistentLocalId)
+                .GetNotRemovedByPersistentLocalId(addressPersistentLocalId)
                 .Remove();
         }
 
         public void CorrectAddressRejection(AddressPersistentLocalId addressPersistentLocalId)
         {
-            StreetNameAddresses
-                .GetByPersistentLocalId(addressPersistentLocalId)
-                .CorrectAddressRejection();
+            var addressToCorrect = StreetNameAddresses
+                .GetNotRemovedByPersistentLocalId(addressPersistentLocalId);
+
+            GuardAddressIsUnique(
```

Comment:
We're going to have to pass the GuardAddressIsUnique method as an action to CorrectRetirement in order to execute the address IsRemoved and Status checks prior to it. See RejectAddress